https/gifts.js

```diff
@@ -16,140 +16,242 @@ const { EVENT } = require("../config/constants");
   }
 );
 
 /**
- * @namespace receiveGift
- * @return {receiveGift~inner} - the returned function
+ * @namespace updateGiftReceivedStatus
+ * @return {updateGiftReceivedStatus~inner} - the returned function
  */
-const receiveGift = functions.https.onCall(
+const updateGiftReceivedStatus = functions.https.onCall(
   /**
-   * marks the gift recieved on the giftees account
+   * Updates the sent status of the gift
    * @inner
    * @param {object} data - details about the giftee
    * @param {string} data.user - user object or uid
-   * @param {boolean} data.value - marks it either true or false
-   * @param {object} [context] - This is used by firebase, no idea what it does, I think its added automatically
+   * @param {string} data.giftId - the uid for the gift
+   * @param {bool} data.isReceived - if the gift is received
    * @returns {Result}
    */
-  async ({ user, value }, context) => {
-    // on the giftee (current user)
-    let gifteeStatus = await markGifteeAccount(
-      { user: user },
-      { field: "received", value: value }
-    );
-
-    let gw2Account = await markGw2Account({
-      user: user,
-      field: "received",
-      value: value
-    });
-
-    // check result and return to frontend
-    if (gifteeStatus.success && gw2Account.success) {
-      return { success: gifteeStatus.success };
-    } else {
-      return { error: gifteeStatus.error };
+  async ({ user, giftId, isReceived }) => {
+    let gameAccountUUID = await getGameAccountUUID(user);
+    if (gameAccountUUID.error) {
+      return { error: "no API key set" };
     }
+
+    let eventDoc = db.collection(CollectionTypes.EVENTS).doc(EVENT);
+
+    let giftDoc = eventDoc
+      .collection(CollectionTypes.EVENTS__GIFTS)
+      .doc(giftId);
+
+    if (!giftDoc.exists) {
+      return { error: `Found no gifts with id: ${giftId}` };
+    }
+
+    let isGiftUpdatedSuccessfully = giftDoc
+      .set(
+        {
+          received: isReceived ? new Date().toISOString() : null
+        },
+        { merge: true }
+      )
+      .then(() => {
+        return true;
+      })
+      .catch(() => {
+        return false;
+      });
+
+    return isGiftUpdatedSuccessfully
+      ? { success: "Successfully updated gift's received status." }
+      : { error: "Failed updating gift's received status." };
   }
 );
 
 /**
- * @namespace reportGift
- * @return {reportGift~inner} - the returned function
+ * @namespace updateGiftReportedStatus
+ * @return {updateGiftReportedStatus~inner} - the returned function
  */
-const reportGift = functions.https.onCall(
+const updateGiftReportedStatus = functions.https.onCall(
   /**
-   * marks the gift reported on the giftees account
+   * Updates the sent status of the gift
    * @inner
    * @param {object} data - details about the giftee
    * @param {string} data.user - user object or uid
-   * @param {boolean} data.value - marks it either true or false
-   * @param {string} [data.message] - message for reporting
```

Comment:
I had this in as there are a number of reasons why a gift may be reported: never got it, undervalued, inappropiate language.

Reply:
Good catch! Will add this.